packages/manager/apps/pci-private-registry/src/api/data/ip-restrictions.ts

```diff
@@ -0,0 +1,107 @@
+import { v6 } from '@ovh-ux/manager-core-api';
+import { aggregateBySpecificKey } from '@/helpers';
+import {
+  FilterRestrictionsServer,
+  TIPRestrictions,
+  TIPRestrictionsDefault,
+  TIPRestrictionsMethodEnum,
+} from '@/types';
+
+const baseUrl = (projectId: string, registryId: string) =>
+  `/cloud/project/${projectId}/containerRegistry/${registryId}/ipRestrictions`;
+
+export const fetchV6URl = async (url: string): Promise<TIPRestrictions[]> => {
+  const { data } = await v6.get(url);
+  return data;
+};
+
+export const getIpRestrictions = async (
+  projectId: string,
+  registryId: string,
+  authorization: FilterRestrictionsServer[],
+) => {
+  const data = await Promise.all(
+    authorization.map(async (auth) => {
+      const getDataServer = await fetchV6URl(
+        `${baseUrl(projectId, registryId)}/${auth}`,
+      );
+      return getDataServer.map((ipBlocks) => ({
+        ...ipBlocks,
+        authorization: auth,
+      }));
+    }),
+  );
+  if (data.length) {
+    const newData = data.flat().map((restriction) => ({
+      ...restriction,
+      draft: false,
+      checked: false,
+      id: restriction.ipBlock,
+    }));
+    return aggregateBySpecificKey(newData, 'ipBlock', 'authorization');
+  }
+  return [];
+};
+
+// Function to iterate through entries and handle each authorization
+export const processIpBlock = async (
```

Comment:
As a global improvement, a function should not take so many arguments as parameters, you can consider refactor it using a context object.